librz/util/subprocess.c

```diff
@@ -125,6 +125,19 @@ static LPWCH override_env(const char *envvars[], const char *envvals[], size_t e
 	return ret;
 }
 
+static void remove_cr_on_cond(char *str) {
+	char *start = str;
+	while (*str) {
+		if (str[0] == '\r' &&
+		    (str[1] == '\n' ||
+		     !(str - start >= 4 && *(str - 4) == '\x1b') /* HACK */)) {
```

Comment:
???

Reply:
Well that condition checks whether the character 4 spaces back is an escape char, and does not remove the `\r` if so ... it's marked `/* HACK */` because the condition shouldn't really be needed, but there seem to be cases where `\r` pops up in the output unexpectedly (e.g. https://ci.appveyor.com/project/rizinorg/rizin/builds/37565774/job/pe7srqflun29gnjm#L4825).

----

Why 4 spaces? Because the escape sequence that needs protection right now is
https://github.com/rizinorg/rizin/blob/154f34f8f1da6ac0223e143c54d4d0da4bbf25c5/librz/include/rz_cons.h#L584 but there are 2 other escape sequences that are similar
https://github.com/rizinorg/rizin/blob/154f34f8f1da6ac0223e143c54d4d0da4bbf25c5/librz/include/rz_cons.h#L583-L585

Reply:
Let's see if I got the problem right:
rz_subprocess originally tried to remove `\r` so that `\r\n` becomes just `\n`, right?

Reply:
Why is the `str[1] == '\n'` condition needed? It's because the `/* HACK */` condition protects too much.

Reply:
> Let's see if I got the problem right:
rz_subprocess originally tried to remove \r so that \r\n becomes just \n, right?

That I think was the original intention, but just doing https://github.com/rizinorg/rizin/pull/512/commits/5bb007faaa1908ab0b823de4bb231e360a261727#diff-04d59abbf789e0c515594c6e252c1c2edec7975d252e1d59d6b767049db87d52R128-R137 resulted in the hot mess of https://ci.appveyor.com/project/rizinorg/rizin/builds/37567072/job/4bqkfj8f006p8pu9#L4083

Reply:
I see... Do you have any idea why? I'm not a big fan of hacks, as you know :D
Maybe some `rz_str_ansi*` function can help? Like `rz_strstr_ansi`?

Reply:
> Do you have any idea why?

No unfortunately ... looks like a major investigation in itself

> Maybe some `rz_str_ansi*` function can help? Like `rz_strstr_ansi`?

ok more precise code is 82b3c2436be1493633e8a37b81ae8e57545070c0 using `strncmp`